### PR TITLE
cucumber-expressions: convert from junit v4 to junit v5

### PR DESCRIPTION
**Pull request by nhojpatrick**
## Summary

JUnit v5 jupiter update only

## Details

Change from junit v4 to junit v5.
Also fix hamcrest meta project dependency, changing to org.hamcrest:hamcrest:2.1.

## How Has This Been Tested?

`mvn clean install` for using both jdk 8 and 11

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [x] The change has been ported to Java.
- [ ] The change has been ported to Ruby.
- [ ] The change has been ported to JavaScript.
- [ ] The change has been ported to Go.
- [ ] The change has been ported to .NET.
- [x] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by nhojpatrick** [comment]
regarding assertThrows i've broken most into using testMethod to try and break up potential huge blocks of code into hopefully easier to read chunks

i've also change some lambdas into old style because the classname appears in error messages

i've used hamcrest is(equalTo(xxx)) so the full error message is check, as some of the previous message checking was doing a contains type check, so want to make sure the whole message was checked.

File: cucumber-expressions/java/pom.xml [hunk 1]
```diff
@@ -17,29 +17,32 @@
 
     <properties>
         <project.Automatic-Module-Name>io.cucumber.cucumberexpressions</project.Automatic-Module-Name>
+        <maven.compiler.source>1.8</maven.compiler.source>
+        <maven.compiler.target>1.8</maven.compiler.target>
```
Comment (mpkorstanje) [inline cucumber-expressions/java/pom.xml]: I don't think its needed but it doesn't do any harm

Comment (aslakhellesoy) [reply cucumber-expressions/java/pom.xml]: Not needed because it's defined `cucumber-parent` right?

File: cucumber-expressions/java/pom.xml [hunk 2]
```diff
@@ -17,29 +17,32 @@
...
     </properties>
 
     <scm>
         <connection>scm:git:git://github.com/cucumber/cucumber-expressions-java.git</connection>
         <developerConnection>scm:git:git@github.com:cucumber/cucumber-expressions-java.git</developerConnection>
         <url>git://github.com/cucumber/cucumber-expressions-java.git</url>
-        <tag>HEAD</tag>
     </scm>
 
+    <build>
```
Comment (aslakhellesoy) [inline cucumber-expressions/java/pom.xml]: This is already defined in https://github.com/cucumber/cucumber-parent/blob/85620180cb6d6a802dd42bb0c5bb19ad6793bab4/pom.xml#L179-L188

Can we remove this?

Comment (mpkorstanje) [reply cucumber-expressions/java/pom.xml]: Yeah. Should come from the latest parent pom.

**Pull request merged**